spec/requests/trashbin_api_spec.rb

```diff
@@ -74,73 +73,137 @@
     let(:resource_kind) { trashed_resource.kind }
     let(:resource_id) { trashed_resource.id }
 
-    let(:payload) {{
-      parent: {
-        kind: parent_kind,
-        id: parent_id
-      }
-    }}
+    context 'without payload' do
+      subject { put(url, headers: headers) }
 
-    it_behaves_like 'an identified resource' do
-      let(:resource_id) { "doesNotExist" }
-    end
+      context 'resource in root of the project' do
+        it_behaves_like 'an identified resource' do
+          let(:resource_id) { "doesNotExist" }
+        end
 
-    it_behaves_like 'an identified resource' do
-      let(:resource_id) { untrashed_resource.id }
-    end
+        it_behaves_like 'an identified resource' do
+          let(:resource_id) { untrashed_resource.id }
+        end
 
-    it_behaves_like 'an identified resource' do
-      let(:parent_id) { "doesNotExist" }
-      let(:resource_class) { Folder }
-    end
+        it_behaves_like 'a client error' do
```

Comment:
This test would read a bit better if it were wrapped in `context 'with deleted project'`, with the before block nested in the context instead of the shared example.